lib/bridge/IrcHandler.js

```diff
@@ -367,6 +391,44 @@ IrcHandler.prototype.onPart = Promise.coroutine(function*(req, server, leavingUs
         req.log.info("No mapped matrix rooms for IRC channel %s", chan);
         return;
     }
+
+    // Presence syncing
+    //  When an IRC user quits, debounce before leaving them from matrix rooms. In the meantime,
+    //  update presence to "offline". If the user rejoins a channel before timeout, do not part
+    //  user from the room. Otherwise timeout and leave rooms.
+
+    if (kind === "quit") {
+        // TODO: This should be replaced with "disconnected" as per matrix-appservice-irc#222
+        let presence = "offline";
+        yield Promise.all(matrixRooms.map((room) => {
+            req.log.info(
+                "Setting presence of %s in room %s to %s",
+                matrixUser.getId(), room.getId(), presence
+            );
+            return this.ircBridge.getAppServiceBridge().getIntent(
+                matrixUser.getId()
+            ).client.setPresence(presence);
+        }));
```

Comment:
This doesn't make sense. Presence is user-scoped, not room-scoped, so why are we setting the presence N times, one for each mapped room?